Add disk buffer tempfile cleanup

codevbus opened the pull request:
### Description:
Extends automatic cleanup to orphaned temp files from DiskBufferReader

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

Commit:
add tempfile creation

- break PID retrieval into sep. function

Commit:
add tmpfile cleanup func

Commit:
add file cleanup to main cleanup func

codevbus commented:
Setting this to draft, will need to merge this: https://github.com/trufflesecurity/disk-buffer-reader/pull/9 before this functionality can be implemented.

Commit:
refactor file logic to only return name string

Commit:
add temp buffer naming to gcs

Commit:
add temp buffer naming to s3

Commit:
add temp buffer naming to filesystem

Commit:
add temp buffer naming to git

rosecodym commented on pkg/cleantemp/cleantemp.go:
```diff
@@ -85,3 +102,45 @@ func CleanTempDir(ctx logContext.Context, dirName string, pid int) error {
 	return nil
 }
+
+func CleanTempFile(ctx logContext.Context, fileName string, pid int) error {
```
Does this method differ from `CleanTempDir` in any way other than inverting the `file.IsDir()` check? (Other than its log messages.) If not, and it's always called alongside `CleanTempFile`, then can they just be collapsed into one function that deletes things irrespective of dir-ness?

Put another way: Instead of adding a new function to delete files and always calling it alongside the old function that deletes directories, could you modify the old function to delete files and directories together? That would result in less code to maintain without creating any new coupling.

Commit:
consolidate cleanup functions

- have single function handle both files and dirs
- remove interface(not needed with a single func implementation)
- change calls to `New(...)` to reflect config implementation
- simplify automation in main.go
- update disk-buffer-reader dependency

Commit:
Merge branch 'main' into add-disk-buffer-tempfile-cleanup

dustin-decker commented on pkg/cleantemp/cleantemp.go:
```diff
@@ -27,84 +41,62 @@ func MkdirTemp() (string, error) {
+			// Check if the name matches any live PIDs
 			for _, pidval := range pids {
-				if strings.Contains(dir.Name(), fmt.Sprintf("-%s-", pidval)) {
+				if strings.Contains(file.Name(), pidval) {
```
can you revert this change?

Commit:
integrate changes from pr #2133

Commit:
Merge branch 'main' into add-disk-buffer-tempfile-cleanup

dustin-decker commented:
Looks like there is a merge conflict still

codevbus commented:
@dustin-decker - this looks like a net new merge conflict after additional changes were made to the underlying pre-merge changes: https://github.com/trufflesecurity/trufflehog/pull/2136

ahrav committed:
merge main

ahrav committed:
merge main

Commit:
Merge branch 'main' into add-disk-buffer-tempfile-cleanup

Commit:
checkout from main to revert conflict issues

Commit:
re-add buffer logic to git

codevbus commented on pkg/cleantemp/cleantemp.go:
```diff
@@ -27,27 +26,33 @@ func MkdirTemp() (string, error) {
+// is generally handled by "github.com/trufflesecurity/disk-buffer-reader"
+func MkFilename() string {
+	pid := os.Getpid()
+	filename := fmt.Sprintf("%s-%d-", "trufflehog", pid)
```
Could you elaborate? Currently `MkFilename()` gets called whenever a diskbuffer file needs to be generated.

Commit:
interface no longer needed

Commit:
move string format to global const

Commit:
Merge branch 'main' into add-disk-buffer-tempfile-cleanup

codevbus commented on pkg/cleantemp/cleantemp.go:
```diff
@@ -14,11 +14,16 @@ import (
 	logContext "github.com/trufflesecurity/trufflehog/v3/pkg/context"
 )
 
+const (
```
@rosecodym - moved the string/formatting to a global const. I think leaving https://github.com/trufflesecurity/trufflehog/pull/2130/files#diff-60ea9ceab29f368c394ac493e6abe8a9195a1fbf7ea2dedbe420f94b8e84cfcfR78 as is may still make sense as it should catch anything with a live pid value in it even if, for whatever reason, it didn't originate from this module.

Commit:
Merge branch 'main' into add-disk-buffer-tempfile-cleanup
